src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Configuration/TargetFrameworkProjectConfigurationDimensionProvider.cs

```diff
@@ -31,16 +31,15 @@ public TargetFrameworkProjectConfigurationDimensionProvider(IProjectXmlAccessor
         {
             Requires.NotNull(project, nameof(project));
 
-            string targetFrameworkProperty = await _projectXmlAccessor.GetEvaluatedPropertyValue(project, ConfigurationGeneral.TargetFrameworkProperty).ConfigureAwait(true);
-            if (targetFrameworkProperty != null)
+            string targetFrameworksProperty = await _projectXmlAccessor.GetEvaluatedPropertyValue(project, ConfigurationGeneral.TargetFrameworksProperty).ConfigureAwait(true);
+            if (targetFrameworksProperty != null)
             {
-                // If the project already defines a specific "TargetFramework" to target, then this is not a cross-targeting project and we don't need a target framework dimension.
-                return ImmutableArray<string>.Empty;
+                return BuildUtilities.GetPropertyValues(targetFrameworksProperty);
             }
             else
             {
-                string targetFrameworksProperty = await _projectXmlAccessor.GetEvaluatedPropertyValue(project, ConfigurationGeneral.TargetFrameworksProperty).ConfigureAwait(true);
-                return BuildUtilities.GetPropertyValues(targetFrameworksProperty);
+                // If the project doesn't have a "TargetFrameworks" property, then this is not a cross-targeting project and we don't need a target framework dimension.
+                return ImmutableArray<string>.Empty;
```

Comment:
The base type does a similar check for the dimension's property so in this case it'll check for TargetFramework but we need to do the check for TargetFrameworks. So we need to override it or pass in a property name which seems not any clearer than just overriding it.